ocaml/typing/jkind.mli

```diff
@@ -455,27 +480,36 @@ type sub_or_intersect =
 
 (** [sub_or_intersect t1 t2] does a subtype check, returning a [sub_or_intersect];
     see comments there for more info. *)
-val sub_or_intersect : t -> t -> sub_or_intersect
+val sub_or_intersect : l -> r -> sub_or_intersect
 
 (** [sub_or_error t1 t2] does a subtype check, returning an appropriate
     [Violation.t] upon failure. *)
-val sub_or_error : t -> t -> (unit, Violation.t) result
+val sub_or_error : l -> r -> (unit, Violation.t) result
+
+(** Like [sub], but returns the subjkind with an updated history.
+    Pre-condition: the super jkind must be fully settled; no variables
+    which might be filled in later. *)
+val sub_jkind_l : l -> l -> (l, Violation.t) result
```

Comment:
Yes, it *is* strange! But it's something we need in the module inclusion check. So there can be variables on the right, but they won't be filled in later.

Reply:
I actually don’t think this is strange at all. It is completely standard to have a check that is left less than left, where the second left is not being inferred. And it should be a left, not a left/right. This operation is called _subsumption_ in Stephen’s thesis, as distinct from the usual _subtyping_ ordering that is left-vs-right. In addition to being left-vs-left, subsumption is also between “schemes” and should allow polymorphism to be instantiated — we don’t have those two concepts right now, but we will soon enough.

I think it would be worth giving this operation a different name here to make the distinction with subtyping clearer.

Reply:
I accept the intuition that, if the second left is fixed and won't change in the future, then this is OK.